### PR TITLE
style: added conditional text to /available/; Closes: #817

### PR DESCRIPTION
**Pull request by 9921622**
I didn't do a test for the text that pops up for hidden quests because I still have no idea what constitutes as an "available" quest.

However, I made sure to manually test everything, and it seems everything works as intended

### CODE REVIEW

File: src/quest_manager/templates/quest_manager/quests.html [hunk 1]
```diff
@@ -71,6 +71,14 @@
...
         <p>You have not joined a course yet for this semester.</p>
         <p><a href="{% url 'courses:create' %}" class="btn btn-info" role="button">Join a Course</a></p>
+      {% elif not available_quests.exists %}
+        {% if in_progress_submissions.exists %}
+          <p>You have no new quests available, but you can find some quests you have already started in your 'In Progress' tab above.</p>
+        {% elif request.user.profile.num_hidden_quests > 0 %}
+          <p>You have no new quests available, but you do have hidden quests which you can view by hitting the 'Show Hidden Quests' button above.</p>
+        {% elif awaiting_submission %}
+          <p>New quests will become available after your teacher approves your submission!</p>
+        {% endif %}
```
Comment (tylerecouture) [inline src/quest_manager/templates/quest_manager/quests.html:81]: There should be a final {% else %} here with: "There are currently no new quest available to you!"

File: src/quest_manager/tests/test_views.py [hunk 2]
```diff
@@ -159,6 +161,97 @@ def test_start(self):
...
+
+        response = self.client.get(url)
+        self.assertContains(response, "No quests are available.")
+        flake8string = "You have no new quests available, but you can find some quests you have already started in your 'In Progress' tab above."
+        self.assertContains(response, flake8string)
+        QuestSubmission.objects.all().delete()
+        Quest.objects.all().delete()
+        
+        # # #quest should be in 'hidden' ===
+        # quest = Quest.objects.create(
```
Comment (tylerecouture) [inline src/quest_manager/tests/test_views.py:211]: Remove all this commented out code, or put a big comment on top explaining why it's here and what it's waiting for before it can be used.

File: src/quest_manager/tests/test_views.py [hunk 3]
```diff
@@ -159,6 +161,97 @@ def test_start(self):
         # the view should have redirect to the same submission:
         self.assertRedirects(response, sub.get_absolute_url())
 
+    @tag("do")
```
Comment (tylerecouture) [inline src/quest_manager/tests/test_views.py:164]: haha, remove the tag and import!

File: src/quest_manager/tests/test_views.py [hunk 4]
```diff
@@ -159,6 +161,97 @@ def test_start(self):
...
+        """
+        url = reverse('quests:available')
+        user = self.test_student1
+
+        # login
+        success = self.client.login(username=user.username, password=self.test_password)
+        self.assertTrue(success)
+
+        # condition to see any of the 3
+        self.assertTrue(user.profile.has_current_course and not user.is_staff)
```
Comment (tylerecouture) [inline src/quest_manager/tests/test_views.py:187]: Excellent comment.  I know it seems small and silly, but allowed me to instantly understand why you were asserting this.

File: src/quest_manager/tests/test_views.py [hunk 5]
```diff
@@ -159,6 +161,97 @@ def test_start(self):
...
+
+        # make sure no available quests
+        Quest.objects.all().delete()
```
Comment (tylerecouture) [inline src/quest_manager/tests/test_views.py:190]: lol, harsh, but fair.

File: src/quest_manager/tests/test_views.py [hunk 6]
```diff
@@ -159,6 +161,97 @@ def test_start(self):
...
+
+        # conditions for msg to show
+        inprogress = QuestSubmission.objects.all_not_completed(user, blocking=True)  # should exist
+        available = Quest.objects.get_available(user, True)  # should not exist
+        self.assertTrue(inprogress.exists()) 
+        self.assertFalse(available.exists())
+
+        response = self.client.get(url)
+        self.assertContains(response, "No quests are available.")
+        flake8string = "You have no new quests available, but you can find some quests you have already started in your 'In Progress' tab above."
```
Comment (tylerecouture) [inline src/quest_manager/tests/test_views.py:205]: Why are you calling this variable `flake8string`?  what does this have to do with flake8? should be `expected_text` or something more informative.

File: src/quest_manager/tests/test_views.py (hunk 2 above)
Comment (tylerecouture) [reply src/quest_manager/tests/test_views.py:211]: Actually, maybe just delete this code and add a new issue on github for "Add tests to 'test_student_no_quests_help_text' " and describe the needed missing tests in the issue

File: src/quest_manager/views.py [hunk 7]
```diff
@@ -372,8 +372,10 @@ def quest_list(request, quest_id=None, template="quest_manager/quests.html"):
     # num_inprogress = QuestSubmission.objects.all_not_completed(request.user).count()
     # num_completed = QuestSubmission.objects.all_completed(request.user).count()
 
+    awaiting_submission = QuestSubmission.objects.filter(is_approved=False, is_completed=True).exists()
```
Comment (tylerecouture) [inline src/quest_manager/views.py:375]: What does "awaiting submission" mean?  Did you mean "awaiting approval"?  Also, this is not filtered for the current user, so it will be getting ALL submissions from every user. (I guess this one really needed a test =D

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Great work here, a few nuances and tweaks requested.  Also currently the "awaiting_submission" won't work as intended I suspect

File: src/quest_manager/tests/test_views.py (hunk 6 above)
Comment (9921622) [reply src/quest_manager/tests/test_views.py:205]: renamed to 'help_text'

File: src/quest_manager/views.py (hunk 7 above)
Comment (9921622) [reply src/quest_manager/views.py]: changed name and now added user=request.user to the filter, as well as a test

File: src/quest_manager/tests/test_views.py (hunk 3 above)
Comment (9921622) [reply src/quest_manager/tests/test_views.py]: removed

**Comment by 9921622** [comment]
Added:
- fixed awaiting_submission by passing an extra argument for user in the filter method
- changed awaiting_submission  to awaiting_approval
- added a test that tests awaiting_submission  if it has submissions from different users
- added an extra help text if user completed all quests with nothing left + tests
-

File: src/quest_manager/tests/test_views.py (hunk 2 above)
Comment (9921622) [reply src/quest_manager/tests/test_views.py]: submitted an issue refrencing this: https://github.com/bytedeck/bytedeck/issues/1011

**Review by tylerecouture — APPROVED** [review]
